app/helpers/application_helper.rb

```diff
@@ -138,6 +138,12 @@ def api_pagination(response)
     will_paginate(numbers.paginate(page: response['current_page'], per_page: response['per_page']))
   end
 
+  # rubocop: disable Rails/OutputSafety
+  def safe_text(str)
+    safe_join(Array(str.html_safe))
+  end
+  # rubocop: enable Rails/OutputSafety
```

Comment:
I have grave concerns about this. `html_safe` was replaced with this method and the two are doing polar opposite things.

[html_safe](https://apidock.com/rails/String/html_safe) marks the reference string as *being* safe and returns an instance of a `SafeBuffer`. "It will be inserted into HTML with no additional escaping performed"

[safe_join](https://apidock.com/rails/ActionView/Helpers/OutputSafetyHelper/safe_join), while also returning an HTML safe string (not sure if it is an instance of a `SafeBuffer` or not), also "all items ... are HTML escaped"

The first method *does not* HTML escape, but the second one does. These two things are not equivalent.

I see the [Rubocop documentation](https://www.rubydoc.info/gems/rubocop-rails/2.3.2/RuboCop/Cop/Rails/OutputSafety), but don't know if it's a simple as a blanket replacement.

Reply:
Appreciate the comments on `safe_text` and `html_safe`/`safe_join`.  I was trying to create a single method that would be responsible for calling `html_safe` and then, using `safe_join` any unescaped HTML would then be escaped, addressing any potential malicious string text.
Agreed that it's not as simple as a blanket replacement, we'll back this change out and turn the cop off (`Rails/OutputSafety`) in the `.rubocop.yml` file.

I also agree with removing the dead code (stubbing restful controller methods) from the respective controllers.  Hindsight tells me this code shouldn't have been as it doesn't add much.  I think Rubocop likes the idea of having every controller method defined so long as it is being referenced somewhere else within the controller, it's definitely a convenience but  all that dead code distracts from grokking the real functionality; let's not try and outthink Rubocop here, we'll turn the cop off (`Rails/LexicallyScopedActionFilter`) in `.rubocop.yml`.

Reply:
Re testing:  My test environment runs cleanly, not sure why you'd be seeing the VCR errors (and others) but duly noted, and hope this doesn't start happening consistently.